### DISCUSSION TITLE
Compiling RetroArch with OpenOrbis: Make -f Makefile.ps4

### DISCUSSION THREAD

**Discussion by godreborn** [OpenOrbis/OpenOrbis-PS4-Toolchain#249] [Q&A]
I'm not sure if this is a problem with openorbis or retroarch.  I was going to post on the retroarch github, but there's no "issues" tab.  anyway, for some reason, it fails instantly like it can't find the environmental variable.  it's set correctly, even checked it, OO_PS4_TOOLCHAIN, but it fails:

![image](https://user-images.githubusercontent.com/13548481/142295261-faef1378-6024-4408-892f-c5e0ac58cc22.png)

**Comment by kiwidoggie** [comment]
Which repo are you compiling this from?

**Comment by godreborn** [comment]
https://github.com/OsirizX/RetroArch/tree/ps4_r4

I'm not sure if Makefile.ps4 or Makefile.orbis is the ps4 version.  Makefile.ps4 gives the error above, and I think the file it's looking for is part of the toolchain, which is installed correctly as I was able to compile one of the samples.  Makefile.orbis iirc uses ps4sdk, which I have as environmental variable, but I either can't get the sdk to compile or it's retroarch that's the problem.

**Comment by illusionyy** [comment]
Missing `-i` for OOSDK includes. `-isysroot $(TOOLCHAIN) -isystem $(TOOLCHAIN)/include`

**Comment by illusionyy** [comment]
https://github.com/OpenOrbis/OpenOrbis-PS4-Toolchain/blob/3bf73118ed0bc436689241cde5fe2072d2cd3a1c/samples/graphics/Makefile#L31-L34

**Comment by godreborn** [comment]
I have that, and these are the errors I encounter with Makefile.orbis as well as Makefile.ps4:

godreborn@GODREBORN-LAPTOP:~/ps4_r4$ make -f Makefile.orbis
clang -c -o version_git.o version_git.c   -m64 -DORBIS -DHAVE_LIBORBIS -std=c11 -mcmodel=large -ffreestanding -nostdlib -nostdinc -fno-builtin -fno-stack-protector -O3 -Wall -pedantic -Wno-zero-length-array -Wno-format-pedantic -D__PS4__ -I/usr/local/ps4-payload-sdk/include -I /usr/local/ps4-payload-sdk/include/sce -target x86_64-scei-ps4-elf -fPIE -DRARCH_INTERNAL -DRARCH_CONSOLE -DHAVE_FILTERS_BUILTIN -DHAVE_LIBRETRODB -DHAVE_GIT_VERSION -DGIT_VERSION=66daf89d77 -DHAVE_CONFIGFILE -DHAVE_CC_RESAMPLER -DHAVE_MENU -DHAVE_GFX_WIDGETS -DHAVE_XMB -DHAVE_OZONE -DHAVE_OVERLAY -DHAVE_THREADS -DHAVE_OPENGL -DHAVE_OPENGLES -DHAVE_OPENGLES2 -DHAVE_EGL -DHAVE_GLSL -DHAVE_7ZIP -D_7ZIP_ST -DHAVE_ZLIB -DHAVE_RTGA -DHAVE_RPNG -DHAVE_RJPEG -DHAVE_RBMP -DHAVE_COMPRESSION -I. -Ideps/7zip -Ilibretro-common/include -Ideps/stb -Ilibretro-common/include/compat/zlib -MT version_git.o -MMD -MP -MF version_git.Tdepend
In file included from version_git.c:17:
In file included from ./version_git.h:20:
libretro-common/include/retro_common_api.h:95:10: fatal error: 'inttypes.h' file not found
#include <inttypes.h>
         ^~~~~~~~~~~~
1 error generated.
make: *** [Makefile.orbis:140: version_git.o] Error 1
godreborn@GODREBORN-LAPTOP:~/ps4_r4$ make -f Makefile.ps4
clang -cc1 -triple x86_64-scei-ps4-elf -munwind-tables -std=c11 -DORBIS -D__ORBIS__ -DHAVE_OOSDK -I. -Idefines -Imemory/ps4 -Ideps -Ideps/7zip -Ideps/libz -Ilibretro-common/include -Ideps/stb -Ilibretro-common/include/compat/zlib -Ideps/rcheevos/include -DHAVE_OPENGLES3 -DHAVE_SOCKET_LEGACY -DHAVE_LIBRETRODB -DHAVE_SHADERPIPELINE -DHAVE_GIT_VERSION -DGIT_VERSION=66daf89d77 -DHAVE_DR_MP3 -DHAVE_CONFIGFILE -DHAVE_RUNAHEAD -DHAVE_CC_RESAMPLER -DHAVE_LANGEXTRA -DHAVE_IMAGEVIEWER -DHAVE_MENU -DHAVE_GFX_WIDGETS -DHAVE_RGUI -DHAVE_MATERIALUI -DHAVE_XMB -DHAVE_OZONE -DHAVE_OVERLAY -DHAVE_VIDEO_LAYOUT -DHAVE_STB_FONT -DHAVE_THREADS -DHAVE_OPENGL -DHAVE_OPENGLES -DHAVE_OPENGLES2 -DHAVE_EGL -DHAVE_GLSL -DHAVE_7ZIP -D_7ZIP_ST -DHAVE_ZLIB -DHAVE_CHD -DWANT_SUBCODE -DWANT_RAW_DATA_SECTOR -DHAVE_RTGA -DHAVE_RPNG -DHAVE_RJPEG -DHAVE_RBMP -DHAVE_NETWORKING -DHAVE_NETWORK_CMD -DHAVE_CHEEVOS -DRC_DISABLE_LUA -DHAVE_COMPRESSION -DHAVE_MAIN -DRARCH_INTERNAL -DRARCH_CONSOLE -DHAVE_FILTERS_BUILTIN -DHAVE_XMB -DHAVE_RGUI -DHAVE_OZONE -DHAVE_ONLINE_UPDATER -DHAVE_UPDATE_CORES -DHAVE_UPDATE_ASSETS -emit-obj -O3 -o memory/ps4/user_mem.o memory/ps4/user_mem.c
memory/ps4/user_mem.c:3:10: fatal error: 'orbis/libkernel.h' file not found
#include <orbis/libkernel.h>
         ^~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [Makefile.ps4:186: memory/ps4/user_mem.o] Error 1
